pallets/invulnerables/src/lib.rs

```diff
@@ -142,12 +164,39 @@ pub mod pallet {
                 Error::<T>::TooManyInvulnerables
             );
 
-            // should never fail since `new` must be equal to or shorter than `TooManyInvulnerables`
-            let mut bounded_invulnerables = BoundedVec::<_, T::MaxInvulnerables>::try_from(new)
-                .map_err(|_| Error::<T>::TooManyInvulnerables)?;
+            let mut new_with_keys = Vec::new();
+
+            // check if the invulnerables have associated validator keys before they are set
+            for account_id in &new {
+                // don't let one unprepared collator ruin things for everyone.
+                let validator_key = T::CollatorIdOf::convert(account_id.clone());
```

Comment:
Is the validator_key the same as the "CollatorId"? If not this shouldn't be called `validator_key`.

Reply:
I think we should name it `collator_id`

Reply:
done